dbms/src/Dictionaries/HashedDictionary.h

```diff
@@ -186,6 +193,23 @@ class HashedDictionary final : public IDictionary
             CollectionPtrType<Float64>,
             CollectionPtrType<StringRef>>
             maps;
+        std::variant<
```

Comment:
You probably won't have to change this after adding the collection type parameter, but if you do, consider using a double-macro approach (it has a name I forgot), like `APPLY_FOR_AGGREGATED_VARIANTS` in Aggregator.cpp.

Reply:
>APPLY_FOR_AGGREGATED_VARIANTS

Like this?

```c
#define APPLY_FOR_TYPE(t)
    CollectionPtrType<t>,\
    SparseCollectionPtrType<t>\

  APPLY_FOR_TYPE(UInt8),
  ...
  APPLY_FOR_TYPE(StringRef)
#undef APPLY_FOR_TYPE
```

I was thinking about this, but decided that separate sparse_maps is cleaner, but I don't have strong filing against this.

So the question is do you want `sparse` to be class template argument or not?